hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -1921,9 +1921,15 @@ public boolean shouldPerformMajorCompaction() throws IOException {
         // If we're enqueuing a major, clear the force flag.
         this.forceMajor = this.forceMajor && !request.isMajor();
 
-        // Set common request properties.
-        // Set priority, either override value supplied by caller or from store.
-        request.setPriority((priority != Store.NO_PRIORITY) ? priority : getCompactPriority());
+        if (request.isAfterSplit()) {
```

Comment:
Hmm. I suppose this is fine. We already have `request#isMajor` to inform us if we need to schedule a major compaction.

`request#isSplit` is just as informative as `request#isAfterSplit`, but no strong opinion there.

Reply:
We should log when we are overriding request priority with the reason why, at DEBUG level at least, but INFO makes sense too.

Reply:
Oh, if you accept my above suggestion wrt a constant value for split housekeeping that is not quite MIN_VALUE then we should not override priority if it is already equal to or less than what we need.

Reply:
Agree, we should override priority only if it is higher than constant (MIN_VALUE + 1000).

Reply:
@virajjasani  One Q.
Rather than handling at this layer by checking whether the call is after a split or not,  can we set a higher priority on the compact request, created after the split op?  So then we dont need to set with these split status etc and handle specially at this layer.

Reply:
There are 2 reasons: 
1. `isAfterSplit` is an additional field which could be useful for maybe some other info in future also. 
2. We already have logic to determine priority present here: `request.setPriority((priority != Store.NO_PRIORITY) ? priority : getCompactPriority());`. Nowhere other than this place, are we using priority setter `request.setPriority(int p)`. So this will provide better alignment.